src/implementation/set_tricks.cpp

```diff
@@ -0,0 +1,19 @@
+/**
+ * Author: Václav Volhejn
+ * Description: Functions to get the closest lower element in a set.
+ *  Given $S$, $k$, returns $\max \{x \in S; x \leq k\}$, or $-\infty$
```

Comment:
The semicolon looks quite unusual to me in set-builder context – I think `\mid` or `:` is a lot more common.

Reply:
What I wanted to convey here would be, in Python:
```python
max([x for x in S if x < k])
```
whereas I think that what you're suggesting would translate to
```python
max([x < k for x in S])
```
so perhaps a better way to write it would be
```latex
\{x \mid x \in S \land x \leq k}
```